lib/bloc/pos_catalog/bloc.dart

```diff
@@ -264,6 +265,15 @@ class PosCatalogBloc with AsyncActionsHandler {
     }
   }
 
+  Future _fetchSales(FetchSales action) async {
+    final result = Map<String, Sale>();
+    final futures = action.paymentHashes.map((hash) => _repository
+        .fetchSaleByPaymentHash(hash)
```

Comment:
Here we query for each item while it is better to fetch all sales we need at once.
Even better maybe populate the sale object on the payment item at the account bloc here: https://github.com/breez/breezmobile/blob/master/lib/bloc/account/account_bloc.dart#L671
Right now we only fetch the hashes and we can easily change it to fetch other sale properties (without the sale lines of course).

Reply:
I did apply the changes, looks better now IMHO.
One thing is it uses the `fetchSaleByID` and it loads the sale lines and it is necessary to load because it is used through the `totalAmountInFiat`.

Reply:
Updated to use the new model `SaleSummary`